draft-ietf-gnap-resource-servers.md

```diff
@@ -0,0 +1,502 @@
+
+~~~
+POST /introspect HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Detached-JWS: ejy0...
+
+{
+    "access_token": "OS9M2PMHKUR64TB8N6BW7OZB8CDFONP219RP1LT0",
+    "proof": "httpsig",
+    "resource_server": "7C7C4AZ9KHRS6X63AJAO"
+}
+~~~
+
+
+
+The AS responds with a data structure describing the token's
+current state and any information the RS would need to validate the
+token's presentation, such as its intended proofing mechanism and key
+material. The response MAY include any fields defined in an access
+token response.
+
+~~~
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+    "active": true,
+    "access": [
+        "dolphin-metadata", "some other thing"
+    ],
+    "key": {
+        "proof": "httpsig",
+        "jwk": {
+                "kty": "RSA",
+                "e": "AQAB",
+                "kid": "xyz-1",
+                "alg": "RS256",
+                "n": "kOB5rR4Jv0GMeL...."
+        }
+    }
+}
+~~~
+
+## Registering a Resource Handle {#rs-register-resource-handle}
+
+If the RS needs to, it can post a set of resources as described in the Resource Access Rights section of 
+{{I-D.ietf-gnap-core-protocol}} to the AS's resource registration endpoint.
+
+The RS MUST identify itself with its own key and sign the
+request.
+
+~~~
+POST /resource HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Detached-JWS: ejy0...
+
+{
+    "access": [
+        {
+            "actions": [
+                "read",
+                "write",
+                "dolphin"
+            ],
+            "locations": [
+                "https://server.example.net/",
+                "https://resource.local/other"
+            ],
+            "datatypes": [
+                "metadata",
+                "images"
+            ]
+        },
+        "dolphin-metadata"
+    ],
+    "resource_server": "7C7C4AZ9KHRS6X63AJAO"
+
+}
+~~~
+
+
+
+The AS responds with a handle appropriate to represent the
+resources list that the RS presented.
+
+~~~
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+    "resource_handle": "FWWIKYBQ6U56NL1"
+}
+~~~
+
+
+
+The RS MAY make this handle available as part of a 
+[response](#rs-request-without-token) or as
+documentation to developers.
+
+\[\[ [See issue #117](https://github.com/ietf-wg-gnap/gnap-core-protocol/issues/117) \]\]
+
+
+
+# Deriving a downstream token {#token-chaining}
+
+Some architectures require an RS to act as a client instance and use a derived access
+token for a secondary RS. Since the RS is not the same entity that made the initial grant
+request, the RS is not capable of referencing or modifying the existing grant. As such, 
+the RS needs to request or generated a new token access token for its use at the secondary RS.
+This internal secondary token is issued in the context of the incoming access token.
+
+While it is possible to use a [token format]{#token-format} that allows for the
```

Comment:
#structure?

Reply:
Weird, that shouldn't have compiled with a broken link. Fixed now.